src/query.test.ts

```diff
@@ -122,4 +122,51 @@ describe('.property()', () => {
     let q = Query.g.property({foo: 'bar', bar: undefined})
     expect(q.toString()).toBe("g.property('foo', 'bar')")
   })
-})
\ No newline at end of file
+})
+
+describe('.as()', () => {
+  it('it takes a label as a string', () => {
+    let q = Query.g.V('foo').as('bar')
+    expect(q.toString()).toBe("g.V('foo').as('bar')")
+  })
+})
+
+describe('.by()', () => {
+  it('it takes a label as a string', () => {
+    let q = Query.g.V('foo').by('bar')
+    expect(q.toString()).toBe("g.V('foo').by('bar')")
+  })
+})
+
+describe('.by()', () => {
+  it('it takes a key as a string', () => {
+    let q = Query.g.V('foo').by('bar')
+    expect(q.toString()).toBe("g.V('foo').by('bar')")
+  })
+})
+
+describe('.from()', () => {
+  it('it takes a target as a string', () => {
+    let q = Query.g.V('foo').addE('bar').from('source')
+    expect(q.toString()).toBe("g.V('foo').addE('bar').from('source')")
+  })
+
+  it('it takes a target as a query', () => {
+    let target = Query.g.V('source')
+    let q = Query.g.V('foo').addE('bar').from(target)
+    expect(q.toString()).toBe("g.V('foo').addE('bar').from(g.V('source'))")
+  })
+})
+
+describe('.to()', () => {
+  it('it takes a target as a string', () => {
+    let q = Query.g.V('foo').addE('bar').to('source')
```

Comment:
Super nitpicky: in your test text you say "target", but in the test itself you say "source"